### PR TITLE
Edit item #13

### LINKED ISSUE #13: As an end-operator/admin, I want to be able to edit items
- [ ] Implement User Interface and API request for editing  items (2 weeks)
- [ ] Implement button to edit items on Landing pages and table view
- [ ] Implement window to be pre-populated with field values
- [ ] Implement POST request to add to DB

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

allows the user to edit an item

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

closes #13

### CODE REVIEW

File: cypress/e2e/items.cy.ts [hunk 1]
```diff
@@ -308,4 +308,140 @@ describe('Items', () => {
...
+    cy.findByText('Edit').click();
+
+    cy.findByRole('button', { name: 'Save' }).click();
+
+    cy.findByText('Please edit a form entry before clicking save').should(
+      'exist'
+    );
+  });
+
+  it('should display an error message there is an unknown error', () => {
```
Comment (joelvdavies) [inline cypress/e2e/items.cy.ts:436]:
```suggestion
  it('should display an error message if there is an unknown error', () => {
```

File: src/items/itemDialog.component.tsx [hunk 2]
```diff
@@ -127,7 +128,13 @@ function ItemDialog(props: ItemDialogProps) {
     new Array(parentCatalogueItemPropertiesInfo.length).fill(false)
   );
 
+  const [formError, setFormError] = React.useState(false);
+  const [formErrorMessage, setFormErrorMessage] = React.useState<
+    string | undefined
+  >(undefined);
```
Comment (joelvdavies) [inline src/items/itemDialog.component.tsx:134]: Can these not be the same state?

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Just the query invalidation needs changing

**Comment by joshuadkitenge** [comment]
wait on https://github.com/ral-facilities/inventory-management-system-api/pull/152

**Review by joelvdavies — APPROVED** [review]
As mentioned on https://github.com/ral-facilities/inventory-management-system-api/pull/152#discussion_r1467369075 I think it would be nice to have placeholder text on on the create/edit dialog, but probably best as a separate issue anyway.

**Pull request merged**